pkg/server/plugin/upstreamca/awssecret/awslib.go

```diff
@@ -0,0 +1,50 @@
+package awssecret
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/aws/aws-sdk-go/service/secretsmanager/secretsmanageriface"
+)
+
+type awsClient interface {
+	DescribeInstancesWithContext(aws.Context, *ec2.DescribeInstancesInput, ...request.Option) (*ec2.DescribeInstancesOutput, error)
+	GetInstanceProfileWithContext(aws.Context, *iam.GetInstanceProfileInput, ...request.Option) (*iam.GetInstanceProfileOutput, error)
+}
+
+func readARN(sm secretsmanageriface.SecretsManagerAPI, arn string) (*string, error) {
+
+	req, resp := sm.GetSecretValueRequest(&secretsmanager.GetSecretValueInput{
```

Comment:
`GetSecretValueWithContext` seems like a better choice and allows you to thread through the context which gives us cancellation, if necessary.